docs/content/latest/architecture/docdb-sharding/tablet-splitting.md

```diff
@@ -195,21 +206,94 @@ Tablet UUID                       Range                                 Leader
+
+### Enable automatic tablet splitting
+
+To enable automatic tablet splitting, use the `yb-master` [`--tablet_split_size_threshold_bytes`](../../../reference/configuration/yb-master/#tablet-split-size-threshold-bytes) flag to specify the size when tablets should split.
+
+The lower the value for the threshold size, the more tablets will exist with the same amount of data.
+
+### Example using a YCSB workload with automatic tablet splitting
+
+In the following example, a three-node cluster is created and uses a YCSB workload to demonstrate the use of automatic tablet splitting in a YSQL database. For details on using YCSB with YugabyteDB, see the [YCSB](../../../benchmark/ycsb-jdbc/) section in the Benchmark guide.
+
+1. Create a three-node cluster.
+
+```sh
+./bin/yb-ctl --rf=3 create --num_shards_per_tserver=1 --ysql_num_shards_per_tserver=1 --master_flags '"tablet_split_size_threshold=30000000"' --tserver_flags '"memstore_size_mb=10"'
+```
+
+2. Create a table for workload.
+
+```sh
+./bin/ysqlsh -c "CREATE DATABASE ycsb;"
+./bin/ysqlsh -d ycsb -c "CREATE TABLE usertable (
+           YCSB_KEY TEXT,
+           FIELD0 TEXT, FIELD1 TEXT, FIELD2 TEXT, FIELD3 TEXT,
+           FIELD4 TEXT, FIELD5 TEXT, FIELD6 TEXT, FIELD7 TEXT,
+           FIELD8 TEXT, FIELD9 TEXT,
+           PRIMARY KEY (YCSB_KEY ASC));"
+```
+
+3. Create the properties file for YCSB at `~/code/YCSB/db-local.properties` and add the following content.
+
+```sh
+db.driver=org.postgresql.Driver
+db.url=jdbc:postgresql://127.0.0.1:5433/ycsb;jdbc:postgresql://127.0.0.2:5433/ycsb;jdbc:postgresql://127.0.0.3:5433/ycsb
+db.user=yugabyte
+db.passwd=
+core_workload_insertion_retry_limit=10
+```
+
+4. Start loading data for the workload.
+
+```sh
+~/code/YCSB/bin/ycsb load jdbc -s -P ~/code/YCSB/db-local.properties -P ~/code/YCSB/workloads/workloada -p recordcount=500000 -p operationcount=1000000 -p threadcount=4
+```
+
+5. Monitor the tablets splitting by going to the YugabyteDB Web UI at `http://localhost:7000/tablet-servers` and `http://127.0.0.1:9000/tablets`.
+
+6. Run the workload.
+
+```sh
+~/code/YCSB/bin/ycsb run jdbc -s -P ~/code/YCSB/db-local.properties -P ~/code/YCSB/workloads/workloada -p recordcount=500000 -p operationcount=1000000 -p threadcount=4
+```
+
+7. Get the list of tablets accessed during the run phase of the workload.
+
+```sh
+diff -C1 after-load.json after-run.json | grep tablet_id | sort | uniq
+```
+
+8. The list of tablets accessed can be compared with `http://127.0.0.1:9000/tablets` to make sure no presplit tablets have been accessed.
+
+## Current tablet splitting limitations
+
+Presplitting tablets and automatic tablet splitting are in beta. To follow the work-in-progress on tablet splitting, see [GitHub #1004](https://github.com/yugabyte/yugabyte-db/issues/1004).
```

Comment:
this note here is much more ok, since it's clearly related to limitations, rather than upfront, before the user even gets to try it